portal-impl/src/com/liferay/portlet/internal/PortletURLImpl.java

```diff
@@ -424,6 +424,28 @@ public void setBeanParameter(PortletSerializable portletSerializable) {
 
 		if (Validator.isNull(paramName)) {
 			paramName = portletSerializableClass.getSimpleName();
+
+			// If RenderURL.setBeanParameter(PortletSerializable) is called from
+			// within a bean portlet that uses Spring for IoC, and if Spring
+			// dynamically created the bean at runtime using CGLib, then the
+			// parameter name will have "$$EnhanderBySpringCGLib" as a suffix.
+
+			int pos = paramName.indexOf("$$EnhancerBySpringCGLIB");
```

Comment:
@ngriffin7a "$$EnhancerBySpringCGLIB" and "$Proxy$_$$_WeldClientProxy" are both suffix, not possible to be in the middle of the paramName, right?

Reply:
@shuyangzhou I never observed "$$EnhancerBySpringCGLIB" and "$Proxy$_$$_WeldClientProxy"  in the middle of the paramName. I only observed them as a suffix.